baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -694,6 +693,48 @@ def get_snapshot_from_provider(self, name, url):
         snapshot_block_hash = self.config["snapshots"][name]["block_hash"]
         return (snapshot_file, snapshot_block_hash)
 
+    # check if a given provider has the compatible snapshot
+    # available in its metadata and return the metadata of this
+    # snapshot if it's available
+    def try_fallback_provider(self, name, url):
+        print(f"Getting snapshots' metdata from {name} instead...")
+        self.get_snapshot_metadata(name, url)
+        return self.config["snapshots"].get(name, None)
+
+    # check if some of the providers has the compatible snapshot
+    # available in its metadadata and return the provider name
+    # along with snapshot itself if found
+    def find_fallback_provider(self, providers):
+        for name, url in providers.items():
+            snapshot = self.try_fallback_provider(name, url)
+            if snapshot is not None:
+                return (snapshot, name)
+        return None
+
+    # tries to get the latest compatible snapshot from the given
+    # provider's metadata
+    #
+    # if the snapshot not found, tries to find it in other known
+    # providers
+    def get_snapshot_from_provider_with_fallback(self, name, url):
+        print_and_log(f"Getting snapshots' metadata from {name}...")
+
+        self.get_snapshot_metadata(name, url)
+        snapshot = self.config["snapshots"].get(name, None)
+
```

Comment:
```suggestion
fallback_provider = None
```
otherwise it fails with `UnboundLocalError: local variable 'fallback_provider' referenced before assignment`